feature/navigation/drawer/src/main/kotlin/app/k9mail/feature/navigation/drawer/ui/folder/FolderListItem.kt

```diff
@@ -36,21 +56,44 @@ internal fun FolderListItem(
         },
         badge = {
             FolderListItemBadge(
-                unreadCount = displayFolder.unreadMessageCount,
-                starredCount = displayFolder.starredMessageCount,
+                unreadCount = unreadCount,
+                starredCount = starredCount,
                 showStarredCount = showStarredCount,
+                expandableState = if (treeFolder !== null && treeFolder.children.isNotEmpty()) isExpanded else null,
             )
         },
     )
+
+    // Managing children
+    Column(modifier = Modifier.fillMaxWidth().animateContentSize()) {
+        if (!isExpanded.value) return
+        if (treeFolder === null) return
+        for (child in treeFolder.children) {
+            var displayParent = treeFolder.value
+            var displayChild = child.value
+            if (displayChild === null) continue
+            FolderListItem(
+                displayFolder = displayChild,
+                selected = false,
+                showStarredCount = showStarredCount,
+                onClick = { onClick(displayChild) },
+                folderNameFormatter = folderNameFormatter,
+                modifier = modifier.then(Modifier.padding(start = MainTheme.spacings.triple)),
+                treeFolder = child,
+                parentPrefix = if (displayParent is DisplayAccountFolder) displayParent.folder.name else null,
+            )
+        }
+    }
 }
 
 @Composable
 private fun mapFolderName(
     displayFolder: DisplayFolder,
     folderNameFormatter: FolderNameFormatter,
+    parentPrefix: String? = "",
 ): String {
     return when (displayFolder) {
-        is DisplayAccountFolder -> folderNameFormatter.displayName(displayFolder.folder)
+        is DisplayAccountFolder -> folderNameFormatter.displayName(displayFolder.folder).removePrefix("$parentPrefix/")
```

Comment:
Maybe this could be done when creating the TreeFolder structure.

Reply:
I think this is fine for now and we could address this when we touch the legacy components.